pkg/scheduler/scheduler.go

```diff
@@ -532,6 +520,16 @@ func (sched *Scheduler) scheduleOne() {
 		return
 	}
 
+	// Run "reserve" plugins.
+	for _, pl := range plugins.ReservePlugins() {
+		if err := pl.Reserve(plugins, assumedPod, suggestedHost); err != nil {
+			klog.Errorf("error while running %v reserve plugin for pod %v: %v", pl.Name(), assumedPod.Name, err)
+			sched.recordSchedulingFailure(assumedPod, err, SchedulerError,
+				fmt.Sprintf("reserve plugin %v failed", pl.Name()))
+			metrics.PodScheduleErrors.Inc()
```

Comment:
What happened if first plugin successfully, but second one failed; the plugin may updates some data in it; similar to other phase. IMO, we may give a callback to plugin on failures.

Reply:
From the design [doc](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling/scheduling-framework.md#extension-points), we can revert all reserve operations in the `Reject` extension point if any `Reserve` plugins failed.

For the `Prebind` extension point, we indeed don't have a revert mechanism. I think this problem also exists in current implementation: In the bind phase, scheduler invokes `bindVolumes` and `bind` in order. If `bindVolumes` succeeded but `bind` failed, we do not revert the volume binding operation.

BTW, if all `Reserve` plugins succeeded but the bind phase failed, need we to revert all reserve operations? In the current implementation, we invoke `sched.config.SchedulerCache.ForgetPod` to revert assumed pods in the cache. Maybe we need this mechanism for `Reserve` plugins too.

cc @bsalamat

Reply:
IMO, to simplify, just give an `Event` to plugins on failure at any point :) Plugins should handle those Event accordingly and scheduler handle failure on framework scope.